modules/targets/generic-linux.nix

```diff
@@ -52,6 +52,24 @@ in {
       . "${pkgs.nix}/etc/profile.d/nix.sh"
       . "${profileDirectory}/etc/profile.d/hm-session-vars.sh"
     '';
+    
+    programs.zsh.envExtra = ''
+      # Make system functions available to zsh
+      () {
```

Comment:
It's covered under "anonymous functions" [here](https://zsh.sourceforge.io/Doc/Release/Functions.html) but it's effectively equivalent to:
```
foo() {
...
}
foo
unset -f foo
```
That is it's an anonymous function that is automatically called, and because it's anonymous, doesn't leave behind unneeded functions in your env.

The function itself is needed for `setopt LOCAL_OPTIONS CASE_GLOB EXTENDED_GLOB`, specifically the `LOCAL_OPTIONS` part which lets me set shell options only for the scope of the function.